main/accounts/index.js

```diff
@@ -565,17 +565,81 @@ class Accounts extends EventEmitter {
+      }
+      this.current().requests[handlerId].data.maxFeePerGas = baseFee
+      this.current().update()
+      cb()
+    }
+  }
+
+  setPriorityFee (priorityFee, handlerId, cb) {
+    if (this.invalidValue(priorityFee)) return cb(new Error('Invalid priority fee'))
+    if (!this.current()) return
+    if (this.current().requests[handlerId] && this.current().requests[handlerId].type === 'transaction') {
+      if (parseInt(priorityFee, 'hex') > 9999 * 1e9) priorityFee = '0x' + (9999 * 1e9).toString(16)
+      const baseFee  = this.current().requests[handlerId].data.maxFeePerGas
+      const gasLimit = this.current().requests[handlerId].data.gasLimit
+      const fee = parseInt(baseFee, 'hex') + parseInt(priorityFee, 'hex')
+      const limit = parseInt(gasLimit, 'hex')
+      if (fee * limit > FEE_MAX) {
+        log.warn('Operation would set fee over hard limit')
+        priorityFee = '0x' + (Math.floor(FEE_MAX / limit) - parseInt(baseFee, 'hex')).toString(16)
+      }
+      this.current().requests[handlerId].data.maxPriorityFeePerGas = priorityFee
+      this.current().update()
+      cb()
+    }
+  }
+
+  setGasPrice (gasPrice, handlerId, cb) {
+    if (this.invalidValue(gasPrice)) return cb(new Error('Invalid gas price'))
+    if (!this.current()) return
+    if (this.current().requests[handlerId] && this.current().requests[handlerId].type === 'transaction') {
+      if (parseInt(gasPrice, 'hex') > 9999 * 1e9) gasPrice = '0x' + (9999 * 1e9).toString(16)
+      const gasLimit = this.current().requests[handlerId].data.gasLimit
+      const fee = parseInt(gasPrice, 'hex')
+      const limit = parseInt(gasLimit, 'hex')
+      if (fee * limit > FEE_MAX) {
+        log.warn('Operation would set fee over hard limit')
+        gasPrice = '0x' + Math.floor(FEE_MAX / limit)
+      }
+      this.current().requests[handlerId].data.gasPrice = gasPrice
+      this.current().update()
+      cb()
+    }
+  }
+
+  setGasLimit (limit, handlerId, cb) {
+    if (this.invalidValue(limit)) return cb(new Error('Invalid gas limit'))
     if (!this.current()) return // cb(new Error('No Account Selected'))
     if (this.current().requests[handlerId] && this.current().requests[handlerId].type === 'transaction') {
-      if (parseInt(price, 'hex') > 9999 * 1e9) price = '0x' + (9999 * 1e9).toString(16)
-      const gasLimit = this.current().requests[handlerId].data.gas
-      if (parseInt(price, 'hex') * parseInt(gasLimit, 'hex') > FEE_MAX) {
-        log.warn('setGasPrice operation would set fee over hard limit')
-        price = '0x' + Math.floor(FEE_MAX / parseInt(gasLimit, 'hex')).toString(16)
+      limit = parseInt(limit, 'hex') 
+      if (limit > 12.5e6) limit = 12.5e6
+      const { type, maxFeePerGas, maxPriorityFeePerGas, gasPrice } = this.current().requests[handlerId].data
+      const fee = type === '0x2' ? parseInt(maxFeePerGas, 'hex') + parseInt(maxPriorityFeePerGas, 'hex') : parseInt(gasPrice, 'hex')
```

Comment:
maybe we could at least put all the `type === '0x2'` checks into some sort of shared utility so that we don't need to repeat it everywhere and can document what the check really means?

Reply:
Sure, I think this was the only place that it's used though

Reply:
I think we had a few instances of it already and there's 4 more in this PR. can just see this getting sprinkled all over the code and when the eip-1559 stuff isn't as fresh in the mind (or for other devs) it might get a little confusing

Reply:
Yeah, happy to update this but I'm not sure what pattern you have in mind here

Reply:
it doesn't have to block the merge but maybe something like a function on our Transaction module like `isEip1559Transaction(rawTxData)`